### ISSUE TITLE
Slow Down FPS in Metal 3.2  macOS Sequoia 15.0 

### ISSUE THREAD

**Issue by p-7** [godotengine/godot#97164]
### Tested versions

- Reproducible in : 

Godot Engine v4.4.dev.custom_build.694d3c293 
**Metal 3.2** - Forward+ - Using Device #0: Apple - Apple M2 Pro (Apple8)

- Not Reproducible in : 

1. Godot Engine v4.4.dev.custom_build.694d3c293 , 
   **Metal 3.1**- Forward+ - Using Device #0: Apple - Apple M2 Pro (Apple8)

2. Godot Engine v4.3.stable.official.77dcf97d8  
    **Vulkan 1.2.283** - Forward+ - Using Device #0: Apple - Apple M2 Pro

### System information

macOS Sequoia 15.0 , Apple M2 Pro

### Issue description

After upgrading macOS to  -> macOS Sequoia 15.0  
All running projects  slow down in Metal 2.3  : 60fps -> 15fps

1. In new system and Metal 3.2  running project decrease FPS frame rate cyclic : 
60 -> 15fps …  5sec works 60fps … 5sec 15fps … and repeat
editor : Godot Engine v4.4.dev.custom_build.694d3c293

2. In old system and Metal 3.1 works the same projects OK  60fps all time 
editor:  Godot Engine v4.4.dev.custom_build.694d3c293

3. In editor : Godot Engine v4.3.stable.official.77dcf97d8
and Vulkan 1.2.283 all projects works OK :
 - in upgraded macOS Sequoia 15  
 - and  in old macOS 14

### Steps to reproduce

1. upgrade macOS to Sequoia 15.0 from old macOS 14 
 … Metal 3.1. change to Metal 3.2  in Godot 4.4  … in 4.3 is Vulkan 1.2

3. open any old project , small scene like example Jolt Physics 
… in Godot 4.4 slow down  cyclic for 5sec , 

### Minimal reproduction project (MRP)

Open small project  like Exmaples Jolt Physics , open any your old projects.

**Comment by akien-mga** [comment]
CC @stuartcarnie

**Comment by stuartcarnie** [comment]
I have been running Godot with Metal on Sequoia since the first public beta, and have not yet seen a regression in frame rates running any of my usual test projects. If you have an MRP, that would be helpful.

In the meantime, can you run your `v4.4.dev.custom_build.694d3c293` version using Vulkan. To do this, open a terminal and `cd` to the test project. Run the following, to start the Godot editor with Vulkan:

```sh
<path_to_godot>/Godot.app/Contents/Macos/Godot --rendering-device vulkan --editor --print-fps
```

To run the game, drop the `--editor` argument:

```sh
<path_to_godot>/Godot.app/Contents/Macos/Godot --rendering-device vulkan --print-fps
```

> [!NOTE]
> 
> When Godot reports Metal 3.2, it is showing the maximum supported API on the OS you are running. Godot currently generates Metal 3.0 shader source, regardless of OS version.

**Comment by p-7** [comment]
Hi,

I preparing additional tests :

1. For old project.  Example project, Jolt example : joints.
2. For new cleen project. Preparing new cleen , Jolt  example : joints.

For old project :
 - in macOS 14. all works normall , when run project  and in editor OK
 - in macOS 15 , slow down, jump frames on screen , when run project, in editor OK no jumps


BUT, I found difference in settings : V-Sync : OFF , in old project , ON in new cleen project.

- in macOS 14  , old projects , V-Sync:OFF , I did’t see difference on running project, all works ok

- in macOS 15 , old projects , V-Sync:OFF , „visible” effect is jump, freeze, if running project works, 
  process up to ~50ms ( see on screens )  if I „shake" window with running project all stabilise to ~60fps, 
  procces ~16ms

- in new projects , V-Sync:ON and all stable 60 fps. 


To summarize the test :

 - In macOS 14  if  V-Sync is ON or OFF I don’t see this „visible” effects on running project. 
 - In macOS 15 if V-Sync is OFF , runing project „jump” , „freeze” , see „visible” changes, process up ~50ms



 
![1](https://github.com/user-attachments/assets/d2f54163-c295-4de1-bedb-6cc32133cf08)



![2](https://github.com/user-attachments/assets/f3767963-75d7-41fe-b208-90eb93c21f45)



Thanks! 
Maybe it's nothing special :)
